src/instructlab/data/generate.py

```diff
@@ -37,11 +37,18 @@
     default=config.DEFAULT_CHUNK_WORD_COUNT,
     show_default=True,
 )
+# TODO - DEPRECATED - Remove in a future release
 @click.option(
     "--num-instructions",
     type=click.INT,
-    help="Number of instructions to generate.",
-    default=config.DEFAULT_NUM_INSTRUCTIONS,
+    default=-1,
+    hidden=True,
+)
+@click.option(
+    "--sdg-scale-factor",
+    type=click.INT,
+    help="Number of instructions to generate for each seed example. The examples map to sample q&a pairs for new skills. For knowledge, examples are generated with both the sample q&a pairs, as well as chunks of the knowledge document(s), so the resulting data set is typically larger for a knowledge addition for the same value of `--sdg-scale-factor`.",
```

Comment:
This is a very helpful message, but I worry about how it would look when printed out against other ones. Maybe not in this PR, but would it make sense to condense this and then move the more thorough explanation elsewhere, then have this point there?

Reply:
click handles the formatting pretty well.

```
Usage: ilab data generate [OPTIONS]

  Generates synthetic data to enhance your example data

Options:
  --model TEXT                Name of the model used during generation.
                              [default: models/merlinite-7b-lab-Q4_K_M.gguf]
  --num-cpus INTEGER          Number of processes to use.  [default: 10]
  --chunk-word-count INTEGER  Number of words to chunk the document  [default:
                              1000]
  --sdg-scale-factor INTEGER  Number of instructions to generate for each seed
                              example. The examples map to sample q&a pairs
                              for new skills. For knowledge, examples are
                              generated with both the sample q&a pairs, as
                              well as chunks of the knowledge document(s), so
                              the resulting data set is typically larger for a
                              knowledge addition for the same value of `--sdg-
                              scale-factor`.  [default: 30]
  --taxonomy-path PATH        Path to
                              https://github.com/instructlab/taxonomy.git
                              clone or local file path.  [default: taxonomy]
  --taxonomy-base TEXT        Base git-ref to use when generating new
                              taxonomy.  [default: origin/main]
  --output-dir PATH           Path to output generated files.
  --rouge-threshold FLOAT     Threshold of (max) Rouge score to keep samples;
                              1.0 means accept all samples.  [default: 0.9]
  --quiet                     Suppress output of synthesized instructions.
  --endpoint-url TEXT         Custom URL endpoint for OpenAI-compatible API.
                              Defaults to the `ilab model serve` endpoint.
  --api-key TEXT              API key for API endpoint. [default:
                              config.DEFAULT_API_KEY]
  --yaml-rules PATH           Custom rules file for YAML linting.
  --server-ctx-size INTEGER   The context size is the maximum number of tokens
                              the server will consider.  [default: 4096]
  --tls-insecure              Disable TLS verification.
  --tls-client-cert PATH      Path to the TLS client certificate to use.
  --tls-client-key PATH       Path to the TLS client key to use.
  --tls-client-passwd TEXT    TLS client certificate password.
  --model-family TEXT         Force model family to use when picking a
                              generation template
  --help                      Show this message and exit.
```